### PR TITLE
feat(linter): eslint-plugin-unicorn prefer-dom-node-text-content(style)

### PR DESCRIPTION
**Pull request by Ken-HH24**
Try to implement [prefer-dom-node-text-content](https://github.com/sindresorhus/eslint-plugin-unicorn/blob/main/docs/rules/prefer-dom-node-text-content.md) for #684 .

### CODE REVIEW

File: crates/oxc_linter/src/rules/unicorn/prefer_dom_node_text_content.rs [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+
+declare_oxc_lint!(
+    /// ### What it does
+    ///
+    /// Enforces the use of `.textContent` over `.innerText` for DOM nodes.
+    ///
+    /// ### Why is this bad?
+    ///
+    /// There are some advantages of using .textContent,
+    /// like performance and more predictable behavior when updating it.
```
Comment (camc314) [inline crates/oxc_linter/src/rules/unicorn/prefer_dom_node_text_content.rs:29]: maybe consider expanding this based on the stack overflow post/blog post.

something  like

> [innerText](https://developer.mozilla.org/en-US/docs/Web/API/HTMLElement/innerText) returns the visible text contained in a node, while [textContent](https://developer.mozilla.org/en-US/docs/Web/API/Node/textContent) returns the full text, this means that `innerText is much more performance-heavy as it requires layout information to return the result.
> 
> [innerText](https://developer.mozilla.org/en-US/docs/Web/API/HTMLElement/innerText) is defined only for HTMLElement objects, while [textContent](https://developer.mozilla.org/en-US/docs/Web/API/Node/textContent) is defined for all Node objects.

Comment (Ken-HH24) [reply crates/oxc_linter/src/rules/unicorn/prefer_dom_node_text_content.rs]: Thanks for your suggestion. I made some adjustments to the description.

File: crates/oxc_linter/src/rules/unicorn/prefer_dom_node_text_content.rs [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+declare_oxc_lint!(
+    /// ### What it does
+    ///
+    /// Enforces the use of `.textContent` over `.innerText` for DOM nodes.
+    ///
+    /// ### Why is this bad?
+    ///
+    /// There are some disadvantages of using .innerText.
+    /// - `.innerText` is much more performance-heavy as it requires layout information to return the result.
+    /// - `.innerText` is defined only for HTMLElement objects, while `/textContent` is defined for all Node objects.
```
Comment (camc314) [inline crates/oxc_linter/src/rules/unicorn/prefer_dom_node_text_content.rs:30]:
```suggestion
    /// - `.innerText` is defined only for HTMLElement objects, while `.textContent` is defined for all Node objects.

Comment (Ken-HH24) [reply crates/oxc_linter/src/rules/unicorn/prefer_dom_node_text_content.rs]: Thanks!

**Review by camc314 — APPROVED** [review]
Thanks!

**Pull request merged**